homeassistant/components/bayesian/config_flow.py

```diff
@@ -0,0 +1,526 @@
+    }
+).extend(OPTIONS_SCHEMA.schema)
+
+OBSERVATION_BOILERPLATE = vol.Schema(
+    {
+        vol.Required(CONF_P_GIVEN_T): vol.All(
+            selector.NumberSelector(
+                selector.NumberSelectorConfig(
+                    mode=selector.NumberSelectorMode.SLIDER,
+                    step=1.0,
+                    min=0,
+                    max=100,
+                    unit_of_measurement="%",
+                ),
+            ),
+            vol.Range(
+                min=0,
+                max=100,
+                min_included=False,
+                max_included=False,
+                msg="extreme_prob_given_error",
+            ),
+        ),
+        vol.Required(CONF_P_GIVEN_F): vol.All(
+            selector.NumberSelector(
+                selector.NumberSelectorConfig(
+                    mode=selector.NumberSelectorMode.SLIDER,
+                    step=1.0,
+                    min=0,
+                    max=100,
+                    unit_of_measurement="%",
+                ),
+            ),
+            vol.Range(
+                min=0,
+                max=100,
+                min_included=False,
+                max_included=False,
+                msg="extreme_prob_given_error",
+            ),
+        ),
+        vol.Required(CONF_NAME): selector.TextSelector(),
+    }
+)
+
+ADD_ANOTHER_BOX_SCHEMA = vol.Schema({vol.Optional("add_another"): cv.boolean})
+
+STATE_SUBSCHEMA = vol.Schema(
+    {
+        vol.Required(CONF_ENTITY_ID): selector.EntitySelector(
+            selector.EntitySelectorConfig(
+                domain=[SENSOR_DOMAIN, BINARY_SENSOR_DOMAIN, INPUT_BOLEAN_DOMAIN]
+            )
+        ),
+        vol.Required(CONF_TO_STATE): selector.TextSelector(
+            selector.TextSelectorConfig(
+                multiline=False, type=selector.TextSelectorType.TEXT, multiple=False
+            )  # ideally this would be a state selector context-linked to the above entity.
+        ),
+    },
+).extend(OBSERVATION_BOILERPLATE.schema)
+
+NUMERIC_STATE_SUBSCHEMA = vol.Schema(
+    {
+        vol.Required(CONF_ENTITY_ID): selector.EntitySelector(
+            selector.EntitySelectorConfig(
+                domain=[SENSOR_DOMAIN, INPUT_NUMBER_DOMAIN, NUMBER_DOMAIN]
+            )
+        ),
+        vol.Optional(CONF_ABOVE): selector.NumberSelector(
+            selector.NumberSelectorConfig(
+                mode=selector.NumberSelectorMode.BOX, step="any"
+            ),
+        ),
+        vol.Optional(CONF_BELOW): selector.NumberSelector(
+            selector.NumberSelectorConfig(
+                mode=selector.NumberSelectorMode.BOX, step="any"
+            ),
+        ),
+    },
+).extend(OBSERVATION_BOILERPLATE.schema)
+
+
+TEMPLATE_SUBSCHEMA = vol.Schema(
+    {
+        vol.Required(CONF_VALUE_TEMPLATE): selector.TemplateSelector(
+            selector.TemplateSelectorConfig(),
+        ),
+    },
+).extend(OBSERVATION_BOILERPLATE.schema)
+
+
+def _convert_percentages_to_fractions(
```

Comment:
I am not entirely sure what these are doing but we could just store it as the user inputs it instead of converting.
We could do that conversion once when loading the entry instead (in the platform in `setup_entry`).

Reply:
I tried to make this work but couldn't
```
    # convert from percentages to fractional probabilities
    prior: float = config[CONF_PRIOR] / 100
    probability_threshold: float = config[CONF_PROBABILITY_THRESHOLD] / 100
    for observation in observations:
        observation[CONF_P_GIVEN_T] = observation[CONF_P_GIVEN_T] / 100
        observation[CONF_P_GIVEN_F] = observation[CONF_P_GIVEN_F] / 100
```
The only way to convert from a percentage to a fraction is to divide by 100, however there is no obvious way of checking if it has already been done. Therefore any repeated calls to `async_setup_entry` duplicates the division. This particularly becomes an issue when using options flows etc.

Solutions I see:
1. Keep as is
2. Abandon trying to use percentages in options flow and switch to using probabilities of 0 to 1, although I think percentages are more intuitive for user-facing UIs, it may keep things simple for users that switch from YAML or find old tutorials online
3. Create some new `Percentage` type and pass that to `async_setup_entry()` and only convert from that type to `float`